src/cider/examples/CiderRuntimeModuleExample.cpp

```diff
@@ -45,21 +45,15 @@ class UserAllocator : public CiderAllocator {
 
 int main(int argc, char** argv) {
   // generate for expression "a * b + b"
-  SubstraitExprBuilder* builder = new SubstraitExprBuilder();
-  ::substrait::NamedStruct* schema =
-      SubstraitExprBuilder::makeNamedStruct(builder,
-                                            {"a", "b"},
-                                            {CREATE_SUBSTRAIT_TYPE_FULL_PTR(I64, false),
-                                             CREATE_SUBSTRAIT_TYPE_FULL_PTR(I64, false)});
-  ::substrait::Expression* field0 = SubstraitExprBuilder::makeFieldReference(0);
-  ::substrait::Expression* field1 = SubstraitExprBuilder::makeFieldReference(1);
-  ::substrait::Expression* multiply_expr = SubstraitExprBuilder::makeExpr(
-      builder, "multiply", {field0, field1}, CREATE_SUBSTRAIT_TYPE_FULL_PTR(I64, false));
-  ::substrait::Expression* add_expr =
-      SubstraitExprBuilder::makeExpr(builder,
-                                     "add",
-                                     {multiply_expr, field1},
-                                     CREATE_SUBSTRAIT_TYPE_FULL_PTR(I64, false));
+  SubstraitExprBuilder builder({"a", "b"},
+                               {CREATE_SUBSTRAIT_TYPE_FULL_PTR(I64, false),
+                                CREATE_SUBSTRAIT_TYPE_FULL_PTR(I64, false)});
+  ::substrait::Expression* field0 = builder.makeFieldReference(0);
+  ::substrait::Expression* field1 = builder.makeFieldReference("b");
+  ::substrait::Expression* multiply_expr = builder.makeScalarExpr(
+      "multiply", {field0, field1}, CREATE_SUBSTRAIT_TYPE_FULL_PTR(I64, false));
```

Comment:
How to know the full supported function name list?

Reply:
It will be done in the expression level validator through same function lookup as plan level validator. The supported functions are listed in [yamls](https://github.com/intel/BDTK/blob/main/src/cider/function/internals/).

Reply:
Assuming users need to do this check by themselves? Validator API can help here.

Reply:
The validation API will be called first when try to offloading expr eval and will fallback if fails.  This is planed to be done at frontend side. `if(validate(exprs)) { do offloading; }`